commands/clone.go

```diff
@@ -101,12 +101,6 @@ func transformCloneArgs(args *Args) {
 					}
 				}
 
-				if !isSSH &&
-					args.Command != "submodule" &&
-					!github.IsHttpsProtocol() {
-					isSSH = repo.Private || repo.Permissions.Push
```

Comment:
Why did this logic go away? I imagine it should still be relevant if someone is explicitly configuring hub to use protocols other than `https` (if they want to keep old behavior)

Reply:
Hi @mislav, sorry for the delay. School's been busy!

This would be taken care of by `Project.GitURL`. If the `hub.protocol` is set to `ssh` or `git`, this would override the defaulting-to-`https` and use the user-defined protocol. Removing this block of logic just prevents the URLs from defaulting to `ssh` even if the repo is private or you have push permission. I felt that this would be in line with the spirit of #1596 for first-time users of `hub` to be able to use their GCM without any surprises.